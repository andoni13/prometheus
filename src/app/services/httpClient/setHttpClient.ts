import {HTTPClient, HTTPClientImpl} from 'app/services/httpClient/types';
import {Error} from 'app/services/Error/types';

/**
 * Abstract function to initialize an httpClient
 * **/

const setHttpClient = (
	httpClient: HTTPClientImpl,
	errorService: (error: Error) => void
): HTTPClient => ({
	fetchData(endpoint: string, options?: RequestInit) {
		return httpClient.fetchData(endpoint, errorService, options);
	},
	//post, put, delete... implementations
});

export default setHttpClient;
