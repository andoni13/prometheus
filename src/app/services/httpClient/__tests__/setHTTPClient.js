import setHttpClient from '../setHttpClient';

describe('Set HttpClient', () => {
	test('httpClient methods snapshot', () => {
		const httpClient = setHttpClient({});
		expect(httpClient).toMatchSnapshot({
			fetchData: expect.any(Function),
		});
	});

	test('httpClient returns methods from injected client', () => {
		const client = {
			fetchData: jest.fn(),
		};

		const errorService = jest.fn();
		const httpClient = setHttpClient(client, errorService);
		const endpoint = 'test';
		const options = {};

		httpClient.fetchData(endpoint, options);
		expect(client.fetchData).toHaveBeenCalledWith(
			endpoint,
			errorService,
			options
		);
		expect(client.fetchData).toHaveBeenCalledTimes(1);
	});
});
