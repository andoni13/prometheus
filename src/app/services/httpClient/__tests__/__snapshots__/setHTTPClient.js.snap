// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Set HttpClient httpClient methods snapshot 1`] = `
Object {
  "fetchData": Any<Function>,
}
`;
